src/chttpd_cors.erl

```diff
@@ -306,6 +316,35 @@ get_cors_config(#httpd{cors_config = undefined}) ->
 get_cors_config(#httpd{cors_config = Config}) ->
     Config.
 
+cors_config(Host, Key, Default) ->
+    config:get(cors_section(Host), Key,
+                     config:get("cors", Key, Default)).
+
+cors_section(Host0) ->
+    {Host, _Port} = split_host_port(Host0),
```

Comment:
The function would work, the config wouldn't. In real life value for header `Host` set to [target's host](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.23), not URI, so cors config would be expected to be under `[cors:partner.com]` where partner.com is the name of our vhost and `http://restricted.dev:8000`is the address of the server that hosts javascript that does ajax call.